src/components/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { Errors } from '../types/Errors';
+
+type Props = {
```

Comment:
use interface prefer. 
Types have unique features but the interface is best for scaling.